lib/utils/batcher/state.go

```diff
@@ -51,16 +58,35 @@ type StateConfig struct {
 	// OnExitOverloaded is called when transitioning from Overloaded to Normal state.
 	// This callback is invoked when batch size drops below threshold/2.
 	//
-	// WARN: For simplicity there is not timer-based transition back to Normal state,
-	// it only happens when batch size drops below threshold/2.
-	// So if the last batch is larger and there are no new events, the state will remain Overloaded.
 	// Is not provided, no action is taken on state transition.
```

Comment:
I would document that `OnExitOverloaded` might be called by a background goroutine rather than as a result of calling `UpdateState`, right now both `OnEnterOverloaded` and `OnExitOverloaded` are only called synchronously.